zypp/sat/detail/PoolImpl.cc

```diff
@@ -505,8 +505,24 @@ namespace zypp
 	  // Add removed locales+fallback except they are still in current
 	  for ( Locale lang: localesTracker.removed() )
 	  {
-	    for ( ; lang && ! localeIds.current().count( IdString(lang) ); lang = lang.fallback() )
-	    { localeIds.removed().insert( IdString(lang) ); }
+	    if ( lang && ! localeIds.current().count( IdString(lang) ) ) {
+
+	      //remove the requested one
+	      localeIds.removed().insert( IdString(lang) );
+
+	      //remove fallbacks
+	      const auto &currLangs = localeIds.current();
+	      auto checkIsFallback = [&lang]( const IdString &currLocStr_r ){
+		Locale currLoc(currLocStr_r);
+		return ( currLoc && Locale(currLoc).hasFallback(lang) );
```

Comment:
Doppeltgemoppelt.   `Locale currLoc(currLocStr_r);` and `Locale(currLoc).hasF...`

Actually you don't need the extra `currLoc` variable. Even if `currLangs` had an empty string included, `Locale(currLocStr_r)` would be the empty Locale (Locale::noCode) and `hasFallback` would always be `false`.

>  return Locale(currLocStr_r).hasFallback(lang);

Reply:
Done